src/Args.cs

```diff
@@ -0,0 +1,137 @@
+namespace DynamicProbes;
+
+public enum ArgType // NOTE! Keep in sync with Libstapsdt.ArgType!
+{
+    NoArg  = Libstapsdt.ArgType.NoArg,
+#pragma warning disable CA1720 // Identifier contains type name (by-design for familiarity)
+    UInt8  = Libstapsdt.ArgType.UInt8,
+    Int8   = Libstapsdt.ArgType.Int8,
+    UInt16 = Libstapsdt.ArgType.UInt16,
+    Int16  = Libstapsdt.ArgType.Int16,
+    UInt32 = Libstapsdt.ArgType.UInt32,
+    Int32  = Libstapsdt.ArgType.Int32,
+    UInt64 = Libstapsdt.ArgType.UInt64,
+    Int64  = Libstapsdt.ArgType.Int64,
+#pragma warning disable CA1720 // Identifier contains type name
+}
+
+public interface IArgType
+{
+    static abstract ArgType ArgType { get; }
+}
+
+public interface IFireArgLong
+{
+    long UncheckedValue { get; }
```

Comment:
We always call the function from the C library with the `long` arguments, even if the probe was registered as `Probe<Int8, Int8>`.

I wonder if it's safe to pass the function parameters to C as 4-byte types; and if yes, why allow registering the probe as unit8 in the first place?